sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.batch;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.implementation.util.FluxUtil;
+import com.azure.core.implementation.util.ImplUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.implementation.models.ServicesSubmitBatchResponse;
+import com.azure.storage.blob.models.StorageErrorException;
+import com.azure.storage.blob.models.StorageException;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/*
+ * This class contains helper methods for dealing with batch requests.
+ */
+class BlobBatchHelper {
+    /*
+     * Newline characters used in HTTP
+     */
+    static final String HTTP_NEWLINE = "\r\n";
+
+    /*
+     * This pattern matches finding the "Content-Id" of the batch response.
+     */
+    private static final Pattern CONTENT_ID_PATTERN = Pattern
+        .compile("Content-ID:\\s?(\\d+)", Pattern.CASE_INSENSITIVE);
+
+    /*
+     * This pattern matches finding the status code of the batch response.
+     */
+    private static final Pattern STATUS_CODE_PATTERN = Pattern
+        .compile("HTTP\\/\\d\\.\\d\\s?(\\d+)\\s?\\w+", Pattern.CASE_INSENSITIVE);
+
+    // This method connects the batch response values to the individual batch operations based on their Content-Id
+    static Mono<SimpleResponse<Void>> mapBatchResponse(BlobBatch batch, ServicesSubmitBatchResponse batchResponse,
+        boolean throwOnAnyFailure, ClientLogger logger) {
+        /*
+         * Content-Type will contain the boundary for each batch response. The expected format is:
+         * "Content-Type: multipart/mixed; boundary=batchresponse_66925647-d0cb-4109-b6d3-28efe3e1e5ed"
+         */
+        String contentType = batchResponse.getDeserializedHeaders().getContentType();
+
+        // Split on the boundary [ "multipart/mixed; boundary", "batchresponse_66925647-d0cb-4109-b6d3-28efe3e1e5ed"]
+        String boundary = contentType.split("=", 2)[1];
```

Comment:
I would be proactive about ensuring that the split actually results in 2 items, lest you get an ArrayIndex out of bounds exception.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756